src/Native/LdaNative/lda_engine.cpp

```diff
@@ -129,11 +124,7 @@ namespace lda {
         model_block_.reset(nullptr);
         samplerQueue_.reset(nullptr);
 
-        for (int i = 0; i < num_threads_; ++i)
-        {
-            delete samplers_[i];
-        }
-        delete[] samplers_;
```

Comment:
📝 This block was responsible for some of the crashes observed in CI, and especially the crashes prior to 0cbfa54. The destructor is missing a null check for samplers_.